cmd/git-sync/webhook.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// WebHook structure
+type Webhook struct {
+	// URL for the http/s request
+	URL string
+	// Method for the http/s request
+	Method string
+	// Code to look for when determining if the request was successful.
+	//   If this is not specified, request is sent and forgotten about.
+	Success *int
+	// Timeout for the http/s request
+	Timeout time.Duration
+	// BackoffDuration for failed webhook calls
+	BackoffDuration time.Duration
+}
+
+func (w *Webhook) Do() error {
+	req, err := http.NewRequest(w.Method, w.URL, nil)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), w.Timeout)
+	defer cancel()
+	req = req.WithContext(ctx)
+
+	//
```

Comment:
I agree with this statement, but maybe it doesn't need to be said?

Reply:
Seems that this is pointing at an incorrect line now? There is an empty comment here-- so I will remove that. But I assume this was meant for another line.